feat(self): add ptfx player effects

Primexz opened the pull request:
idk some ptfx effects for our local player.
i don't know if the whole thing makes sense but here we go.

Primexz committed:
feat(self): add ptfx player effects

Primexz committed:
fix(ptfx): fix indents

Yimura reviewed (dismissed):
Looks good, waiting for @Seanghost117 to add their changes.

Seanghost117 committed:
feat(self): ptfx improvements (#3)

* feat(self): add ptfx player effects

* fix(ptfx): fix indents

* feat(self): updated ptfx player effect

- Added More PTFX
- Allow users to change the asset type and select a effect
- Removed enums and switch case
- Added variables to store the asset name and effect name
- Updated the logic because there was a problem with switching asset tabs

---------

Co-authored-by: Primexz <johann@pepebot.info>
Co-authored-by: Johann <76482511+Primexz@users.noreply.github.com>

Primexz committed:
Merge branch 'master' into ptfx

Primexz committed:
Merge branch 'ptfx' of github.com:Primexz/YimMenu into ptfx

Yimura commented on src/backend/looped/self/ptfx.cpp:
```diff
@@ -0,0 +1,39 @@
+
+        void show_ptfx_effect(const char* fx_name, const char* name)
+        {
+			for (const auto& ptfx_bone : ptfx_bones)
```
mixed indentation (tabs and spaces)

Seanghost117 replied:
Forgot to fix that when I made my edits, I’ll let @Primexz handle that one, don’t wanna Pr off a Pr again

Primexz replied:
I am currently working on the idea of @Seanghost117 .
I will fix the indents in the commit as well

Yimura reviewed (changes requested):
Overall good code just one mixed indentation.

Primexz committed:
feat(ptfx): more friendly effect names (not perfect yet..)

Primexz committed:
Revert "feat(ptfx): more friendly effect names (not perfect yet..)"

This reverts commit e3d6b74b7e4c35fdccfda4eeb6dd6a6ecbce360a.

Primexz committed:
fix(ptfx): fix indents
